src/usrp_drivers/utils/driveroptions.cpp

```diff
@@ -20,121 +20,160 @@ DriverOptions::DriverOptions() {
+    std::map<uint32_t, uint32_t> rx_intf_antenna_to_channel_map;    // Maps rx intf antenna number to channel number
+    std::map<uint32_t, uint32_t> tx_main_antenna_to_channel_map;    // Maps tx main antenna number to channel number
 
     // Get number of physical antennas
-    main_antenna_count_ = boost::lexical_cast<uint32_t>(
+    auto main_antenna_count = boost::lexical_cast<uint32_t>(
                                 config_pt.get<std::string>("main_antenna_count"));
-    intf_antenna_count_ = boost::lexical_cast<uint32_t>(
+    auto intf_antenna_count = boost::lexical_cast<uint32_t>(
                                 config_pt.get<std::string>("intf_antenna_count"));
 
+    uint32_t device_num = 0;    // Active device counter
+
     // Iterate through all N200s in the json array
     for (auto n200 = n200_list.begin(); n200 != n200_list.end(); n200++)
     {
         std::string addr = "";
-        bool rx = false;
-        bool tx = false;
-        bool rx_int = false;
-        std::string main_antenna = "";
-        std::string intf_antenna = "";
+        std::string rx_channel_0 = "";
+        std::string rx_channel_1 = "";
+        std::string tx_channel_0 = "";
+
         // Iterate through all N200 parameters and store them in variables
         for (auto iter = n200->second.begin(); iter != n200->second.end(); iter++)
         {
             auto param = iter->first;
             if (param.compare("addr") == 0) {
                 addr = iter->second.data();
             }
-            else if (param.compare("rx") == 0) {
-                rx = (iter->second.data().compare("true") == 0);
+            else if (param.compare("rx_channel_0") == 0) {
+                rx_channel_0 = iter->second.data();
             }
-            else if (param.compare("tx") == 0) {
-                tx = (iter->second.data().compare("true") == 0);
+            else if (param.compare("rx_channel_1") == 0) {
+                rx_channel_1 = iter->second.data();
             }
-            else if (param.compare("rx_int") == 0) {
-                rx_int = (iter->second.data().compare("true") == 0);
-            }
-            else if (param.compare("main_antenna") == 0) {
-                main_antenna = iter->second.data();
-            }
-            else if (param.compare("intf_antenna") == 0) {
-                intf_antenna = iter->second.data();
+            else if (param.compare("tx_channel_0") == 0) {
+                tx_channel_0 = iter->second.data();
             }
             else {
                 throw std::invalid_argument("Invalid N200 parameter in config file");
             }
         }
 
-        // If current n200 is transmitting, receiving, or receiving from interferometer, add to devices
-        if (tx || rx || rx_int)
+        // If current n200 is connected to an antenna on any channel, add to devices
+        if (rx_channel_0.compare("") != 0 || rx_channel_1.compare("") != 0 || tx_channel_0.compare("") != 0)
         {
```

Comment:
I feel like there's gotta be a way to simplify the logic in this code a bit, but I'm not a c++ expert so I can't give any specific advice. Punching it into chatgpt gave the following simplification though:
```
void processChannel(const std::string& channel, const std::string& type, uint32_t device_num, uint32_t main_antenna_count, uint32_t intf_antenna_count, std::unordered_map<uint32_t, uint32_t>& antenna_to_channel_map) {
    if (!channel.empty()) {
        if (channel.substr(0, 1) == type) {
            uint32_t antenna_num = boost::lexical_cast<uint32_t>(channel.substr(1));
            if ((type == "m" && antenna_num >= main_antenna_count) || (type == "i" && antenna_num >= intf_antenna_count)) {
                throw std::invalid_argument("Invalid antenna number for " + type + " array");
            }
            antenna_to_channel_map[antenna_num] = device_num * 2 + (type == "i" ? 1 : 0);
        } else {
            throw std::invalid_argument("Cannot parse N200 " + channel + " array identifier");
        }
    }
}

// In your main function
if (!rx_channel_0.empty() || !rx_channel_1.empty() || !tx_channel_0.empty()) {
    devices_ += ",addr" + std::to_string(device_num) + "=" + addr;

    processChannel(rx_channel_0, "m", device_num, main_antenna_count, intf_antenna_count, rx_main_antenna_to_channel_map);
    processChannel(rx_channel_1, "m", device_num, main_antenna_count, intf_antenna_count, rx_main_antenna_to_channel_map);
    processChannel(tx_channel_0, "m", device_num, main_antenna_count, intf_antenna_count, tx_main_antenna_to_channel_map);

    ++device_num;
}
```
I feel a little dirty using ai to give suggestions, but it looks like there might be some nice simplifications in there.

Reply:
I don't think that code is actually correct (if `channel` is not `"m"`, then it will throw an error), but I can definitely try tidying it up with a function. It's a good starting point, at any rate

Reply:
Actually yes there are several definite errors in the code above. I expected better from chatGPT